### PR TITLE
Added file size upper bound to request range

### PR DESCRIPTION
**Pull request by hintofbasil**
The range header is often rounded up above the max file size due to rounding to the nearest chunk size.  If a server is strict about the "Range" header it will return a 216 "Range Not Satisfiable".

This simply caps the upper range to the file size.

### CODE REVIEW

File: src/__tests__/XhrFileReader-test.js [hunk 1]
```diff
@@ -182,13 +182,24 @@ describe("XhrFileReader", function() {
...
       fileReader.loadRange([0, 4], throwOnError(resolve));
       jest.runAllTimers();
     }).then(function(tags) {
       expect(require('xhr2').XMLHttpRequest.setRequestHeader.mock.calls[0][1]).toBe("bytes=0-1023");
     });
   });
 
+  pit("should not fetch more than max file size", function() {
+    return new Promise(function(resolve, reject) {
+	  fileReader._size=10;
```
Comment (aadsm) [inline src/__tests__/XhrFileReader-test.js]: Can you fix the indentation? It should be 2 spaces.

**Comment by aadsm** [comment]
Awesome, thank you for this! Just had a small comment.

**Comment by hintofbasil** [comment]
The tab character was introduced from a copy-paste.  I've removed the original too.

**Comment by aadsm** [comment]
Accepted in @1934e6b906b646c4b22fcdc2b8f1a275298b5778

**Pull request closed without merging**